File: pgm/ABC_Java/ABC_249_A.java
import java.io.*;
// import java.math.*;
import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        int[] arr = Arrays.stream (reader.readLine().split (" "))
                        .mapToInt (Integer::parseInt)
                        .toArray();
        final String Takahashi = "Takahashi";
        final String Aoki = "Aoki";
        final String Draw = "Draw";
        final int A = arr[0];
        final int B = arr[1];
        final int C = arr[2];
        final int D = arr[3];
        final int E = arr[4];
        final int F = arr[5];
        final int X = arr[6];
        final Integer takahashiDistance =
            B * A * (X / (A + C)) + B * Math.min (A, X % (A + C));
        final Integer aokiDistance =
            E * D * (X / (D + F)) + E * Math.min (D, X % (D + F));
        if (takahashiDistance.compareTo (aokiDistance) > 0) {
            System.out.println (Takahashi);
        } else if (takahashiDistance.compareTo (aokiDistance) < 0) {
            System.out.println (Aoki);
        } else {
            System.out.println (Draw);
        }
        return;
    }
}

File: pgm/ABC_R/ABC_249_A.r
con <- file(description = "stdin", open = "r")
arr <- readLines(con = con)
close(con)
Takahashi <- "Takahashi"
Aoki <- "Aoki"
Draw <- "Draw"
A <- strsplit(arr[1], split = " ")[[1]][1] |>
    as.integer()
B <- strsplit(arr[1], split = " ")[[1]][2] |>
    as.integer()
C <- strsplit(arr[1], split = " ")[[1]][3] |>
    as.integer()
D <- strsplit(arr[1], split = " ")[[1]][4] |>
    as.integer()
E <- strsplit(arr[1], split = " ")[[1]][5] |>
    as.integer()
f <- strsplit(arr[1], split = " ")[[1]][6] |>
    as.integer()
X <- strsplit(arr[1], split = " ")[[1]][7] |>
    as.integer()
takahashi_distance <- B * A * (X %/% (A + C)) + B * min(A, X %% (A + C))
aoki_distance <- E * D * (X %/% (D + f)) + E * min(D, X %% (D + f))
if (takahashi_distance > aoki_distance) {
    Takahashi |>
        cat("\n", sep = "")
} else if (takahashi_distance < aoki_distance) {
    Aoki |>
        cat("\n", sep = "")
} else {
    Draw |>
        cat("\n", sep = "")
}
q("no")